src/plugins/dashboard/public/dashboard_container/embeddable/integrations/controls/dashboard_control_group_integration.ts

```diff
@@ -193,7 +193,9 @@ function startSyncingDashboardControlGroup(this: DashboardContainer) {
         )
       )
       .subscribe(({ timeslice }) => {
-        this.updateInput({ timeslice });
+        if (!_.isEqual(timeslice, this.getInputAsValueType().timeslice)) {
```

Comment:
Why is `getInputAsValueType` used instead of just `this.getInput()`?

Reply:
I believe `updateInput` can be pretty slow so this was meant as a performance improvement - this is @thompsongl's bit of code, so hopefully he can confirm :)

Reply:
@Heenawter you'e right that this was a performance improvement. `updateInput` causes the whole state in redux to be replaced which causes unnecessary re-renders. To avoid that I wrapped this in a check.

@nreese the reason I'm using `getInputAsValueType()` here instead of `this.getInput()` is to avoid casting. Since portable dashboards was merged, Dashboard Input is actually a union of `DashboardContainerByValueInput | DashboardContainerByReferenceInput`.  `DashboardContainerByReferenceInput` is pretty much just a saved object id and if you use `getInput` it has no idea which subtype you're using so typescript will throw errors. `getInputAsValueType` just returns the input directly as `DashboardContainerByValueInput`.

This weirdness is removed in https://github.com/elastic/kibana/pull/150121, and after that PR merges we'll be able to just use `getInput` as normal.

Reply:
Woops - misread the initial comment, sorry 🤦 Thanks for clarifying.

Reply:
ok, I was confused because `ReferenceOrValueEmbeddable` types `getInputAsValueType` as `Promise<EmbeddableInput>`. I see now that dashboard container has it method called `getInputAsValueType` that just returns results of getInput. Thanks for clarifying.